cypress-io/cypress
# feat: Improve ignore capability to allow module & import matching

- Closes

### Additional details
* Breaks out `ignoreList` into two config options, allowing users to skip processing problematic modules *or* skip mocking a module in a specific location and use the original. This allows for users to work around both problematic module structures (preventing proxifying) and problematic behaviors (overly restrictive type checking, internal function refs, etc)
* Improve error handling, validate config
* Fix bugs around matching a function to a class constructor
* Fix bugs around function invocation
* Improve logging

### Steps to test
Should be covered by test suite
A more robust example app was provided by a user that we can use to validate, but it requires an Intel machine - LMK if you want to give it a try

### How has the user experience changed?
Minor config changes - fallback behavior of old config is provided so this isn't a breaking change

### PR Tasks

- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
